qcodes/tests/dataset/test_paramspec.py

```diff
@@ -106,18 +123,45 @@ def test_copy(name1, name2, name3):
     ps = ParamSpec(name3, "numeric", depends_on=[ps_indep])
     ps_copy = ps.copy()
 
+    att_names = ["name", "type", "label", "unit",
+                 "_inferred_from", "_depends_on"]
+
     attributes = {}
-    for att in ["name", "type", "label", "unit"]:
+    for att in att_names:
         val = getattr(ps, att)
         valc = getattr(ps_copy, att)
         assert val == valc
         attributes[att] = val
 
     # Modifying the copy should not change the original
-    for att in ["name", "type", "label", "unit"]:
-        setattr(ps_copy, att, attributes[att] + "_modified")
+    for att in att_names:
+        if not att.startswith('_'):
+            setattr(ps_copy, att, attributes[att] + "_modified")
+        else:
+            setattr(ps_copy, att, attributes[att] + ['bob'])
         assert getattr(ps, att) == attributes[att]
 
-    ps_copy.add_depends_on([ps_indep_2])
-    assert ps_copy.depends_on == f"{ps_indep.name}, {ps_indep_2.name}"
-    assert ps.depends_on == f"{ps_indep.name}"
+
+def test_serialize():
+
+    p1 = ParamSpec('p1', 'numeric', 'paramspec one', 'no unit',
+                   depends_on=['some', 'thing'], inferred_from=['bab', 'bob'])
+
+    ser = p1.serialize()
+
+    assert ser['name'] == p1.name
+    assert ser['paramtype'] == p1.type
+    assert ser['label'] == p1.label
+    assert ser['unit'] == p1.unit
+    assert ser['depends_on'] == p1._depends_on
+    assert ser['inferred_from'] == p1._inferred_from
+    assert ser['version'] == p1.version
+
+
+def test_deserialize(version_0_serializations):
+
+    yaml = YAML()
+
+    for ser in version_0_serializations:
+        sdict = {a: b for (a, b) in yaml.load(ser).items()}
```

Comment:
what is the reason for supplying fixtures in yaml format? since only the `deserialize` method is tested, the fixtures can just be dicts, right?

Reply:
Good point. Only the top-level object, the `RunDescriber` should have to care about the specific output (yaml/JSON/excel sheet), all other objects just serialize to a dict.